---
title: "Info 510 solutions to final problems"
author: "AJ"
date: "7/7/2021"
output: html_document
---


#*Problem 1*

#Not using the parse below since there is mixup of values between comumn 3 and 4, even after cleanup. Instead manually rectified the problem and importing the data  

```{r}
library(ggplot2)

r <- read.table("ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/variants/humsavar.txt", header = T, 
				skip = 42, sep = "", fill = T, stringsAsFactors = T, flush = T, nrows=79746)
r=r[-1,]

load("D:/OneDrive - UAB - The University of Alabama at Birmingham/SU2021 INFO 710 Programming with Biological Data/sp2/humsvar.RData")
head(humsavar)
```

###*(A) The top five genes that are mutated in various human disease.*

```{r}
top_genes=table(humsavar$gene)
head(top_genes)

top_genes=as.data.frame(top_genes)

head(top_genes[order(-top_genes$Freq),],5)
```


#*B) plot the frequency distribution of disease variants in human genome across all the genes in the file*
#*C) calculate the average number disease causing mutations across all genes in human genome and mark this number on*
#*the previous plot as veritcal red line.*

```{r}
ggplot(top_genes, aes(x = Freq)) + geom_histogram(bins = 40) + scale_x_log10() +
  
  geom_vline(aes(xintercept = mean(Freq)),col='red',size=2) 
```


#*Problem 2* 



#loading and cleaning up the data (removed last column; appeared to be a discrepancy )
```{r}
library(DESeq2)
library(tidyverse)
PNAS<- read.table("https://github.com/cb2edu/info510/raw/2019/04-RNASeq/data/pnas_expression.txt", header = T)
PNAS=column_to_rownames(PNAS, var = "ensembl_ID")
PNAS=PNAS[-c(8)]
```


#removing zerocounts 

```{r}
PNAS<- PNAS[rowSums(PNAS) >= 7,]
```

#visualizing unnormalized data


```{r}

PNAS_pivot=pivot_longer(PNAS, cols=1:7, names_to = "Sample", values_to = "Counts")


plot1= ggplot(PNAS_pivot, aes(x = Sample, y = Counts)) + geom_boxplot(outlier.shape = NA) +
  coord_cartesian(ylim=c(-1, 220)) + 
  
  labs(title="Count distribution of sample before normaziation")


```

#*A)  Applying upper quartile normalization* 

```{r}
PNAS_uqn <-  apply(PNAS, 2,function(x){quantile(x[x>0], 0.75)});
PNAS_uqn_norm <- t(t(PNAS) / PNAS_uqn);

PNAS_uqn_norm=as.data.frame(PNAS_uqn_norm)

PNAS_uqn_norm_pivot = pivot_longer(PNAS_uqn_norm, cols=1:7, names_to = "Sample", values_to = "Counts")



plot2=ggplot(PNAS_uqn_norm_pivot, aes(x = Sample, y = Counts)) + geom_boxplot(outlier.shape = NA) +
  coord_cartesian(ylim=c(0, 3)) + 
  
  labs(title="Upper quartile normaziation")


```

#*A) Applying Deseq Normalization*

```{r}
coldata <- data.frame(condition=c(rep("C",4), rep("T",3)))
row.names(coldata) <- colnames(PNAS)

dds <- DESeqDataSetFromMatrix(countData = PNAS, colData = coldata, design = ~ condition)
dds <- estimateSizeFactors(dds)
```


```{r cache=TRUE}
#get the sizefactors
sizeFactors(dds)
```

To get the count normalized count table:
```{r cache=TRUE}
PNAS_deseq_norm=counts(dds, normalized=T)
```


#Data transformation for ggplot

```{r}
PNAS_deseq_norm=as.data.frame(PNAS_deseq_norm)


PNAS_deseq_norm_pivot=pivot_longer(PNAS_deseq_norm, cols=1:7, names_to = "Sample", values_to = "Counts")



plot3=ggplot(PNAS_deseq_norm_pivot, aes(x = Sample, y = Counts)) + geom_boxplot(outlier.shape = NA) +
  coord_cartesian(ylim=c(0, 180)) + 
  
  labs(title="Deseq normaziation")

```

#*B) 2x2 multiplot figure*

```{r}
library(egg)
ggarrange(plot1, plot2,plot3)
```

